src/svg-quirks-mode/svg-renderer.js

```diff
@@ -0,0 +1,251 @@
+            let text = textElement.textContent;
+            if (text) {
+                textElement.textContent = '';
+                let lines = text.split('\n');
+                text = '';
+                for (let line of lines) {
+                    let tspanNode = this._createSVGElement('tspan');
+                    tspanNode.setAttribute('x', '0');
+                    tspanNode.setAttribute('dy', '1em');
+                    tspanNode.textContent = line;
+                    textElement.appendChild(tspanNode);
+                }
+            }
+        }
+        // Inject fonts that are needed.
+        // It would be nice if there were another way to get the SVG-in-canvas
+        // to render the correct font family, but I couldn't find any other way.
+        // Other things I tried:
+        // Just injecting the font-family into the document: no effect.
+        // External stylesheet linked to by SVG: no effect.
+        // Using a <link> or <style>@import</style> to link to font-family
+        // injected into the document: no effect.
+        let newDefs = this._createSVGElement('defs');
+        let newStyle = this._createSVGElement('style');
+        let allFonts = Object.keys(fontsNeeded);
+        for (let font of allFonts) {
+            if (FONTS.hasOwnProperty(font)) {
+                newStyle.textContent += FONTS[font];
+            }
+        }
+        newDefs.appendChild(newStyle);
+        this._svgTag.insertBefore(newDefs, this._svgTag.children[0]);
+    }
+
+    /**
+     * Transform the measurements of the SVG.
+     * In Scratch 2.0, SVGs are drawn without respect to the width,
+     * height, and viewBox attribute on the tag. The exporter
+     * does output these properties - but they appear to be incorrect often.
+     * To address the incorrect measurements, we append the DOM to the
+     * document, and then use SVG's native `getBBox` to find the real
+     * drawn dimensions. This ensures things drawn in negative dimensions,
+     * outside the given viewBox, etc., are all eventually drawn to the canvas.
+     * I tried to do this several other ways: stripping the width/height/viewBox
+     * attributes and then drawing (Firefox won't draw anything),
+     * or inflating them and then measuring a canvas. But this seems to be
+     * a natural and performant way.
+     */
+    _transformMeasurements () {
+        // Save `svgText` for later re-parsing.
+        let svgText = this._toString();
+
+        // Append the SVG dom to the document.
+        // This allows us to use `getBBox` on the page,
+        // which returns the full bounding-box of all drawn SVG
+        // elements, similar to how Scratch 2.0 did measurement.
+        let svgSpot = document.createElement('span');
+        document.body.appendChild(svgSpot);
+        svgSpot.appendChild(this._svgTag);
+        // Take the bounding box, and then destroy the element.
+        let bbox = this._svgTag.getBBox();
```

Comment:
If there's there any chance that calling `getBBox()` will throw (either it throws internally or `this._svgTag` doesn't have a `getBBox` property) then it might make sense to put this line in a `try` with the `removeChild` call in `finally`.